### PR TITLE
[#8892] Made it more clear where the existing `connection` parameter comes from.

### PR DESCRIPTION
**Pull request by cwaldbieser**
https://twistedmatrix.com/trac/ticket/8892

It is not really clear from the documentation where the `connection` parameter for `twisted.conch.endpoints.SSHCommandClientEndpoint.existingConnection()` is supposed to come from.  The example program _does_ use `proto.transport.conn`, but calling this out in the documentation makes it more obvious.

### CODE REVIEW

**Comment by glyph** [comment]
Hi @cwaldbieser - is there an accompanying Trac ticket for this?

**Comment by cwaldbieser** [comment]
No.  I actually tried going to trac, but the links to login/register or create a new ticket seemed to be broken [1].

[1] http://twistedmatrix.com/trac/newticket , http://twistedmatrix.com/trac/login

**Comment by glyph** [comment]
> No. I actually tried going to trac, but the links to login/register or create a new ticket seemed to be broken [1].

Neither of those links should be necessary any more; you should just click the 'github login' button at the top right.  (Where'd you find a link to `/trac/login`?  We should find that and clean it up.)

**Comment by glyph** [comment]
Oops, not "neither".  "new ticket" is still necessary, once you're logged in with github :)

**Comment by rodrigc** [comment]
@glyph if you log out of Trac, then click on https://twistedmatrix.com/trac/newticket
this text appears: `You are currently not logged in. You may want to do so now.`
If you click on **do so now**, it takes you to https://twistedmatrix.com/trac/login , but
that doesn't work.

That's a bit confusing.  Would be nice if https://twistedmatrix.com/trac/login could be redirected to https://twistedmatrix.com/trac/github/login

**Comment by glyph** [comment]
@rodrigc Totally agree. I think maybe we should file a bug on braid for this?

**Comment by rodrigc** [comment]
@glyph https://github.com/twisted-infra/braid/issues/224

**Comment by cwaldbieser** [comment]
I figured out the Trac login.  I filed an issue.

**Comment by wsanchez** [comment]
@cwaldbieser: What's the issue #?

**Comment by wsanchez** [comment]
Never mind, found it: https://twistedmatrix.com/trac/ticket/8892

File: docs/conch/howto/conch_client.rst [hunk 1]
```diff
@@ -181,7 +181,11 @@ For use of ``SSHCommandClientEndpoint`` that is intended to be completely autono
 
 It is also possible to run commands (one or more) over an
 already-established connection.  This is done using the alternate
-constructor ``SSHCommandClientEndpoint.existingConnection`` .
+constructor ``SSHCommandClientEndpoint.existingConnection`` .  The
+``connection`` argument to that function can be obtained from the
+the ``conn`` property of the ``transport`` property of the protocol
+instance obtained when connection to an
```
Comment (wsanchez) [inline docs/conch/howto/conch_client.rst]: …when **connecting** to…

File: docs/conch/howto/conch_client.rst [hunk 2]
```diff
@@ -181,7 +181,11 @@ For use of ``SSHCommandClientEndpoint`` that is intended to be completely autono
 
 It is also possible to run commands (one or more) over an
 already-established connection.  This is done using the alternate
-constructor ``SSHCommandClientEndpoint.existingConnection`` .
+constructor ``SSHCommandClientEndpoint.existingConnection`` .  The
+``connection`` argument to that function can be obtained from the
+the ``conn`` property of the ``transport`` property of the protocol
```
Comment (wsanchez) [inline docs/conch/howto/conch_client.rst]: (nit) this might be more simply stated as:
…by accessing `transport.conn` on the protocol…

**Comment by wsanchez** [comment]
reviewed.  Please fix the typo and consider the other comment and resubmit (add review keyword to the trac ticket).

**Comment by tomprince** [comment]
One thing to note is that this argument can be any connected `twisted.conch.ssh.connection.SSHConnection`, not just one obtained via `.newConnection`. Unfortunately, this is poorly documented and there is no straightforward interface for constructing one. It is possible to use the private (thus not covered by the comparability guarantee) `_NewConnectionHelper` to create a connection (see [here](https://github.com/ClusterHQ/flocker/blob/master/flocker/provision/_ssh/_conch.py#L185-L197) for an example usage). There is a [ticket](https://twistedmatrix.com/trac/ticket/6414) to expose this as a sensible API.

**Comment by cwaldbieser** [comment]
I updated the text.  You mentioned a typo-- not sure I am seeing it.

**Comment by wsanchez** [comment]
The typo is "when connection to" should be "when connecting to"

**Comment by wsanchez** [comment]
…and you removed that text.  Looks good.

**Review by wsanchez — APPROVED** [review]

**Comment by wsanchez** [comment]
You need to add a topfile before merging.

**Comment by cwaldbieser** [comment]
So-- what is a "topfile", and how do I add one?

**Comment by glyph** [comment]
@cwaldbieser - technically a "news file" :) but we put them in a directory called "topfiles", so the correct name refuses to stick: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles

**Comment by cwaldbieser** [comment]
I think I have added the "topfile" correctly at this point.

**Comment by glyph** [comment]
Pushed to a Twisted namespace branch to give the buildbots a crack at it.

**Comment by cwaldbieser** [comment]
Since this is just a documentation change, I'm not sure why the build failed on rhel7 ...

**Comment by glyph** [comment]
@cwaldbieser Probably it's just an intermittent failure. I resubmitted the build to see if it will pass on a subsequent run.

**Comment by wsanchez** [comment]
Passed, but out of date again, updating.

**Comment by glyph** [comment]
Once more, with feeling.

**Comment by rodrigc** [comment]
Moved to https://github.com/twisted/twisted/pull/631

**Pull request closed without merging**